src/collections/map.rs

```diff
@@ -13,6 +12,69 @@ use crate::store::*;
 use crate::Result;
 use ed::*;
 
+#[derive(Clone)]
+pub struct MapKey<K> {
+    inner: K,
+    inner_bytes: Vec<u8>,
+}
+
+impl<V> Deref for MapKey<V> {
+    type Target = V;
+
+    fn deref(&self) -> &V {
+        &self.inner
+    }
+}
+
+impl<K> Encode for MapKey<K> {
+    fn encode(&self) -> ed::Result<Vec<u8>> {
+        Ok(self.inner_bytes.clone())
+    }
+
+    fn encode_into<W: std::io::Write>(&self, dest: &mut W) -> ed::Result<()> {
+        match dest.write(self.inner_bytes.as_slice()) {
```

Comment:
This should be `write_all` since `write` can succeed even if not all the bytes have been written yet

Reply:
Change made